From f995aab755b3bc1124e64fa848a9671d883475cd Mon Sep 17 00:00:00 2001
From: Jason Felice <jason.m.felice@gmail.com>
Date: Wed, 26 Jan 2022 14:40:08 -0500
Subject: [PATCH] Set gvproxy path for NixOS

---
 pkg/machine/qemu/machine.go | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/pkg/machine/qemu/machine.go b/pkg/machine/qemu/machine.go
index ab1b6c7df..08973d667 100644
--- a/pkg/machine/qemu/machine.go
+++ b/pkg/machine/qemu/machine.go
@@ -15,7 +15,6 @@ import (
 	"strings"
 	"time"
 
-	"github.com/containers/common/pkg/config"
 	"github.com/containers/podman/v3/pkg/machine"
 	"github.com/containers/podman/v3/pkg/rootless"
 	"github.com/containers/podman/v3/utils"
@@ -648,14 +647,7 @@ func CheckActiveVM() (bool, string, error) {
 // startHostNetworking runs a binary on the host system that allows users
 // to setup port forwarding to the podman virtual machine
 func (v *MachineVM) startHostNetworking() error {
-	cfg, err := config.Default()
-	if err != nil {
-		return err
-	}
-	binary, err := cfg.FindHelperBinary(machine.ForwarderBinaryName, false)
-	if err != nil {
-		return err
-	}
+	binary := "@gvproxy@/bin/gvproxy"
 
 	// Listen on all at port 7777 for setting up and tearing
 	// down forwarding
-- 
2.34.1

